plugins/builds/steps/logs.js

```diff
@@ -6,6 +6,44 @@ const request = require('request');
 const ndjson = require('ndjson');
 const MAX_LINES = 100;
 
+/**
+ * Load all pages that are available
+ * @method loadAllLines
+ * @param  {String}     baseUrl   URL to load from (without the .$PAGE)
+ * @param  {Integer}    linesFrom What line number are we starting from
+ * @return {Promise}              Array of log lines
+ */
+function loadAllLines(baseUrl, linesFrom) {
+    return new Promise((resolve) => {
+        const page = Math.floor(linesFrom / MAX_LINES);
+        const output = [];
+
+        request
+            .get(`${baseUrl}.${page}`)
+            // Parse the ndjson
+            .pipe(ndjson.parse({
+                strict: false
+            }))
+            // Filter down to the lines we care about
+            .on('data', (line) => {
+                if (line.n >= linesFrom) {
+                    output.push(line);
+                }
+            })
+            .on('end', () => resolve(output));
+    }).then((lines) => {
+        const linesCount = lines.length;
+
+        // Load from next log if we got stuff AND we reached the edge of a page
+        if (linesCount > 0 && (linesCount + linesFrom) % MAX_LINES === 0) {
```

Comment:
Would there ever be a case where `linesCount + linesFrom` would exceed `MAX_LINES`?